L1Trigger/Phase2DAQProducer/interface/DTCUnit.h

```diff
@@ -0,0 +1,217 @@
+        DTCType getDTCType() 
+        {
+            // Iterate over the ClusterCollection to find the first matching cluster type
+            for (auto& clusterVector : ClusterCollection)
+            {
+                for (auto& cluster_ : clusterVector)
+                {
+                    for (auto& cluster : cluster_)
+                    {
+                        // Check if the cluster type is Ph2PSP or Ph2PSS
+                        if (cluster.getClusterType() == TrackerGeometry::ModuleType::Ph2PSP || 
+                            cluster.getClusterType() == TrackerGeometry::ModuleType::Ph2PSS)
+                        {
+                            return DTCType::PS;
+                        }
+                        // Check if the cluster type is Ph2SS
+                        else if (cluster.getClusterType() == TrackerGeometry::ModuleType::Ph2SS)
+                        {
+                            return DTCType::S2;
+                        }
+                    }
+                }
+            }
+
+                        // If no matching cluster type is found, return Unknown
+            return DTCType::Unknown;
+        }
+
+        std::vector<std::vector<Cluster>>& getClustersOnSLink(const int& index)
+        {
+            return ClusterCollection[index];
+        }
+
+        std::vector<FEDRawData>& getSLinks()
+        {
+            return SLinkCollection;
+        }
+
+        void convertToRawData(const std::size_t index)
+        {
+            std::vector<std::vector<Cluster>>& SLinks_0 = ClusterCollection[index];
+            std::vector<std::vector<Cluster>> newClusterCollection(36);
+
+            // Organize clusters into CIC-0 and CIC-1 clusters
+            for (size_t i = 0; i < SLinks_0.size(); ++i) {
+                std::vector<Cluster>& clusters = SLinks_0[i];
+                std::vector<Cluster> cicId_0_clusters;
+                std::vector<Cluster> cicId_1_clusters;
+
+                for (auto& cluster : clusters) {
+                    if (cluster.getCicId() == 0) {
+                        cicId_0_clusters.push_back(cluster);
+                    } else if (cluster.getCicId() == 1) {
+                        cicId_1_clusters.push_back(cluster);
+                    }
+                }
+
+                newClusterCollection[2 * i] = cicId_0_clusters;
+                newClusterCollection[2 * i + 1] = cicId_1_clusters;
+            }
+
+            // We will store the final 32-bit words (both offset map and payload)
+            std::vector<uint32_t> words;
+
+            // fill the first 4 words with the header with all 1s
+            words.push_back(0xFFFFFFFF);
+            words.push_back(0xFFFFFFFF);
+            words.push_back(0xFFFFFFFF);
+            words.push_back(0xFFFFFFFF);
+
+            // --- Step 1: Create the offset map ---
+            std::vector<uint16_t> offsetMap(36, 0);  // 36 channels, initially 0
```

Comment:
Magic numbers in code not allowed. Define named constants, perhaps all in one place, so that the name gives people a clue what the numbers represent.

Reply:
done